engine/src/main/java/org/pentaho/di/www/StatusServletUtils.java

```diff
@@ -34,7 +35,9 @@
   public static final String STATIC_PATH = "/static";
   public static final String PENTAHO_ROOT = "/pentaho";
 
-  public static String getPentahoStyles() {
+  public static String getPentahoStyles( HttpServletRequest request ) {
+    String root = request.getRequestURI() == null ? StatusServletUtils.PENTAHO_ROOT : request.getRequestURI().substring( 0, request.getRequestURI().indexOf( GetStatusServlet.CONTEXT_PATH ) );
```

Comment:
Please use a variable for `uri` so you don't repeat `request.getRequestURI()` so many times.
This would make it easier to read such a long line.